src/os/bluestore/BlueStore.cc

```diff
@@ -3675,10 +3686,25 @@ void BlueStore::ExtentMap::reshard(
     }
     dout(30) << " extent " << *e << dendl;
 
-    // disfavor shard boundaries that span a blob
-    bool would_span = (e->logical_offset < max_blob_end) || e->blob_offset;
-    if (estimate &&
-	estimate + extent_avg > target + (would_span ? slop : 0)) {
+    bool make_shard_here = false;
+    if (onode_data_has_boundaries) {
+      if (e->blob_start() >= next_boundary) {
+	// this it the place we want to have shard boundary
+	if ((estimate >= target /*we have enough already*/) ||
```

Comment:
As we could neglect shard size growing above - wouldn't it be better to apply segment size boundary unconditionally here as well? I.e. always enforce shard at this boundary irrespective of estimate value?

Reply:
The deal is that size of encoding varies hugely on presence of checksums.
For 4K checksums I get 5.5K of metadata per 4MB.
For 2K checksums I get 3.5K of metadata per 4MB.
For no checksums I get 1.5K of metadata per 4MB.
Maybe the preferred segment size should be dependent on checksum type, but sharding should be fixed?
**I am actually for fixed shard size.**

Reply:
UPDATE:
All test on random write 
split = 4k/16:8k/10:12k/9:16k/8:20k/7:24k/7:28k/6:32k/6:36k/5:40k/5:44k/4:48k/4:52k/4:56k/3:60k/3:64k/3
Compressed 50% , no csum = 4443 bytes / 4MB
Compressed 50%, crc32c = 8776 bytes / 4MB
No-compress, no csum = 2686 bytes / 4MB
No-compress, crc32c = 7082 bytes / 4MB